platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/consensus/CandidateWitness.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.platform.consensus;
+
+import static com.swirlds.logging.LogMarker.CONSENSUS_VOTING;
+
+import com.swirlds.common.utility.IntReference;
+import com.swirlds.platform.internal.EventImpl;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/** A wrapper for a witness which holds additional metadata while an election is ongoing */
+public final class CandidateWitness {
+    private static final Logger logger = LogManager.getLogger(CandidateWitness.class);
+    private final EventImpl witness;
```

Comment:
Javadocs on these member variables would be helpful for consensus classes. I know you usually just put the javadocs on the constructor parameters, but several of these at not passed in the constructor. It may seem overkill, but for new devs they are helpful.